src/List/Extra.elm

```diff
@@ -1946,21 +1946,24 @@ All possible combinations will be explored.
 -}
 lift2 : (a -> b -> c) -> List a -> List b -> List c
 lift2 f la lb =
-    la |> andThen (\a -> lb |> andThen (\b -> [ f a b ]))
+    List.foldl (\a aacc -> List.foldl (\b bacc -> f a b :: bacc) aacc lb) [] la
+        |> List.reverse
```

Comment:
I'm trying to do the complexity analysis in my head, which is a bit tricky but makes me notice that the inner `andThen` is unnecessary, it would work presumable just as well with a simple `map`, ie:

```elm
lift2 f la lb = 
   List.concatMap (\a -> List.map (f a) lb) la
```

would be a more intuitive implementation (perhaps without the currying for micro-opts, but you get the idea).

But my off the cuff Big-O analysis is that the original complexity was _O(a²b)_ and now is _O(ab), right?

Reply:
That sounds correct but I haven't fully done the math to be honest

Reply:
oh yeah I was probably confused by the `foldr`, and assumed that the size of the `append` would increase, which is where I got the quadratic term.